docs/docs/tutorial-rag.md

```diff
@@ -0,0 +1,287 @@
+    query_embedding = (
+        openai.embeddings.create(input=query, model="text-embedding-3-small")
+        .data[0]
+        .embedding
+    )
+    similarities = [
+        np.dot(query_embedding, doc_emb)
+        / (np.linalg.norm(query_embedding) * np.linalg.norm(doc_emb))
+        for doc_emb in article_embeddings
+    ]
+    # Get the index of the most similar document
+    most_relevant_doc_index = np.argmax(similarities)
+    return articles[most_relevant_doc_index]
+
+# highlight-next-line
+class RAGModel(Model):
+    system_message: str
+    model_name: str = "gpt-3.5-turbo-1106"
+
+# highlight-next-line
+    @weave.op()
+    def predict(self, question: str) -> dict: # note: `question` will be used later to select data from our evaluation rows
+        from openai import OpenAI
+        context = get_most_relevant_document(question)
+        client = OpenAI()
+        query = f"""Use the following information to answer the subsequent question. If the answer cannot be found, write "I don't know."
+        Context:
+        \"\"\"
+        {context}
+        \"\"\"
+        Question: {question}"""
+        response = client.chat.completions.create(
+            model=self.model_name,
+            messages=[
+                {"role": "system", "content": self.system_message},
+                {"role": "user", "content": query},
+            ],
+            temperature=0.0,
+            response_format={"type": "text"},
+        )
+        answer = response.choices[0].message.content
+        return {'answer': answer, 'context': context}
+
+# highlight-next-line
+weave.init('rag-qa')
+model = RAGModel(
+    system_message="You are an expert in finance and answer questions related to finance, financial services, and financial markets. When responding based on provided information, be sure to cite the source."
+)
+model.predict("What significant result was reported about Zealand Pharma's obesity trial?")
+```
+
+## Evaluating with an LLM Judge
+
+When there aren't simple ways to evaluate your application, one approach is to use an LLM to evaluate aspects of it. Here is an example of using an LLM judge to try to measure the context precision by prompting it to verify if the context was useful in arriving at the given answer. This prompt was augmented from the popular [RAGAS framework](https://docs.ragas.io/). 
+
+### Defining a scoring function
+
+As we did in the [Build an Evaluation pipeline tutorial](/tutorial-eval), we'll define a set of example rows to test our app against and a scoring function. Our scoring function will take one row and evaluate it. The input arguments should match with the corresponding keys in our row, so `question` here will be taken from the row dictionary. `prediction` is the output of the model. The input to the model will be taken from the example based on its input argument, so `question` here too. 
+
+```python
+# highlight-next-line
+@weave.op()
+async def context_precision_score(question, prediction):
```

Comment:
@scottire this should change to `model_output` from v49 I think. Maybe also a quick overview of the data flow in the eval pipeline (from dataset columns, to model pred, and eval scorers) as overview would help?